### PR TITLE
HAL_Linux: switch Disco to using AP_RCProtocol library for RCIN

### PR DESCRIPTION
**Pull request by tridge**
this is the first step to using AP_RCProtocol for most (or all) RC
input on Linux boards. It also fixes a major bug on Disco where the
old SBUS decoder didn't use strict enough timing checks to filter out
bad frames when video capture is running. That led to erratic flight
mode changes and sometimes loss of control.

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
this conflicts with my #10961 - we will need a way to continue to allow arbitrary RC implementation besides uart-based ones.

**Comment by tridge** [comment]
I can combine the two if you like, using the Multi method

**Comment by tridge** [comment]
I've combined the two PRs, and merged the other one

**Comment by lucasdemarchi** [comment]
Ok, that looks great. I'm also thinking on allowing more than one at the same time - and only switch to the other on failure (or when commanded).  But that has other problems like the calibration not applying to both, etc.

File: libraries/AP_HAL_Linux/RCInput_RCProtocol.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    } else {
+        fd_sbus = -1;
+    }
+    if (dev_115200) {
+        fd_115200 = open_115200(dev_115200);
+    } else {
+        fd_115200 = -1;
+    }
+    rcp.init();
+    printf("SBUS FD %d  115200 FD %d\n", fd_sbus, fd_115200);
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/RCInput_RCProtocol.cpp]: leftover debug?

Comment (tridge) [reply libraries/AP_HAL_Linux/RCInput_RCProtocol.cpp]: I deliberately left it in - very helpful when debugging issues to have some minimal info like this

**Comment by tridge** [comment]
I've gone a step further and removed RCInput_SBUS and RCInput_115200

**Comment by lucasdemarchi** [comment]
Looks good. Did you fly the disco with the new decoder yet?

**Comment by tridge** [comment]
only bench tested with RCInput_RCProtocol, tested it with sumd and sbus. Probably fly it this weekend. The bad decode with old sbus decoder was easily reproducible on the bench, and is gone now.
I haven't tested the UDP input yet, sorry!

**Comment by lucasdemarchi** [comment]
@tridge not easy to test the UDP input yet as I have to publish how to get it running on the remote... I asked if it was tested because I want to have a backup RC in case something goes bad with my implementation. I don't fear much about the ardupilot or "stick decoding " side of things, but rather the wifi handling and getting it to reconnect if it loses connection.

**Pull request merged**